### ISSUE TITLE
simultaneous irms both fail (but delete the file)

### ISSUE THREAD

**Issue by bh9** [irods/irods#4200]
### iRODS Version, OS and Version
4.1.12, Ubuntu 12.04

### What did you try to do?
simultaneous irms (in example simulated with `irm thing & irm thing`, but also done on separate hosts with `cssh`)

### Expected behavior
1 irm succeeds, the other fails with `ERROR: rmUtil: rm error for /seq-dev/home/bh9#Sanger1-dev/thing, status = -817000 status = -817000 CAT_UNKNOWN_FILE` and the catalog entry and files are deleted

### Observed behavior (including steps to reproduce, if applicable)
Both irms fail with `ERROR: rmUtil: rm error for /seq-dev/home/bh9#Sanger1-dev/thing, status = -817000 status = -817000 CAT_UNKNOWN_FILE` and the catalog entry and files are deleted

```
bh9@irods-g1-dev:~$ ils -L thing
  bh9               0 root;replicate;red;red2;irods-seq-i08-fg          373 2018-11-30.13:37 & thing
    fdc45f5eb8963c5a53b71a00a998839a    generic    /irods-seq-i08-fg/home/bh9#Sanger1-dev/thing
  bh9               1 root;replicate;green;green3;irods-seq-sr03-ddn-gg07-3-4-5          373 2018-11-30.13:37 & thing
    fdc45f5eb8963c5a53b71a00a998839a    generic    /irods-seq-sr03-ddn-gg07-3-4-5/home/bh9#Sanger1-dev/thing
bh9@irods-g1-dev:~$ irm thing & irm thing
[1] 4992
ERROR: rmUtil: rm error for /seq-dev/home/bh9#Sanger1-dev/thing, status = -817000 status = -817000 CAT_UNKNOWN_FILE
ERROR: rmUtil: rm error for /seq-dev/home/bh9#Sanger1-dev/thing, status = -817000 status = -817000 CAT_UNKNOWN_FILE
[1]+  Exit 3                  irm thing
bh9@irods-g1-dev:~$ echo $?
3
```

**Comment by alanking** [comment]
Here are some tests on 4.3.1, as released.

I only see 1 error when just running `irm`:
```bash
$ ils -l foo 
  rods              0 demoResc          284 2024-02-16.19:47 & foo 
  rods              1 ufs0          284 2024-02-16.19:47 & foo 
$ irm foo & irm foo 
[1] 230491
remote addresses: 172.21.0.3 ERROR: rmUtil: rm error for /tempZone/home/rods/foo, status = -528002 status = -528002 UNIX_FILE_RENAME_ERR, No such file or directory
$ ils -l foo 
remote addresses: 172.21.0.3 ERROR: lsUtil: srcPath /tempZone/home/rods/foo does not exist or user lacks access permission
[1]+  Done                    irm foo 
$ ils -l foo 
remote addresses: 172.21.0.3 ERROR: lsUtil: srcPath /tempZone/home/rods/foo does not exist or user lacks access permission
$ ils /tempZone/trash/home/rods -l
/tempZone/trash/home/rods:
  rods              0 demoResc          284 2024-02-16.19:47 & foo 
  rods              1 ufs0          284 2024-02-16.19:47 & foo 
```
The command does its job, and we would expect one of these to fail anyway.

With the force flag (to bypass trash), a similar behavior is observed:
```bash
$ ils -l foo 
  rods              0 demoResc          284 2024-02-16.19:48 & foo 
  rods              1 ufs0          284 2024-02-16.19:48 & foo 
$ irm -f foo & irm -f foo 
[1] 230516
remote addresses: 172.21.0.3 ERROR: rmUtil: rm error for /tempZone/home/rods/foo, status = -817000 status = -817000 CAT_UNKNOWN_FILE
Level 0: data object 'foo' is unknown
Level 1: data object 'foo' is unknown
[1]+  Exit 3                  irm -f foo 
$ ils -l foo 
remote addresses: 172.21.0.3 ERROR: lsUtil: srcPath /tempZone/home/rods/foo does not exist or user lacks access permission
```
Notice, however, that "Exit 3" appears in the output this time. Again, I'm not sure that this is unexpected.

As a point of comparison, here is `rm`'s behavior:
```bash
$ touch deleteme
$ ls -l deleteme
-rw-rw-r-- 1 irods irods 0 Feb 16 19:55 deleteme
$ rm deleteme & rm deleteme
[1] 230633
rm: cannot remove 'deleteme': No such file or directory
[1]+  Exit 1                  rm deleteme
$ ls -l deleteme
ls: cannot access 'deleteme': No such file or directory
```

And here is `mv`'s behavior, to mirror the `irm` behavior of renaming an object to the trash collection:
```bash
$ touch mvme
$ ls -l mvme
-rw-rw-r-- 1 irods irods 0 Feb 16 19:59 mvme
$ mv mvme trash & mv mvme trash
[1] 230708
mv: cannot stat 'mvme': No such file or directory
$ ls -l mvme 
ls: cannot access 'mvme': No such file or directory
[1]+  Exit 1                  mv mvme trash
$ ls -l trash
-rw-rw-r-- 1 irods irods 0 Feb 16 19:59 trash
```
I think the behavior of `irm` is pretty similar to `rm` and `mv`.

I see a couple of problems that *could* be addressed:
 1. Make `irm` output consistent.
 2. Prevent multiple removals or renames from happening at once via locking mechanism.

For problem 1, this can be done but it feels pretty low priority to me.
For problem 2, I would at this point like to link the issue to our "operation locking" issue: #5742

**Comment by trel** [comment]
Agreed.   So much has changed since 4.1.12.   I nominate this for closing.   Consider opening a new issue for problem 1 as stated.

**Comment by alanking** [comment]
https://github.com/irods/irods/issues/7513 created.

**Comment by alanking** [comment]
Closing this as resolved/invalid in favor of the more targeted #7513.
